src/component/app/RegisteredUserRoute.tsx

```diff
@@ -19,15 +19,16 @@ export function RegisteredUserRoute(props: RegisteredUserRouteProps) {
 
   if (userLoading) return <span>Loading user...</span>;
 
-  if (userError) return <Navigate to={"../signin"} />;
-
-  if (!user) {
-    return <Navigate to={"../profile"} />;
+  const returnTo = window.location.pathname + window.location.search;
+  if (userError) {
+    return <Navigate to="/signin" state={{ returnTo }} />;
+  } else if (!user) {
+    return <Navigate to={"/profile"} state={{ returnTo }} />;
```

Comment:
Visiting a page that requires the user to exist in the database. Go to profile page so the user can be created and save the originally requested page in the router state.